apps/aecore/src/aec_mining.erl

```diff
@@ -107,7 +107,8 @@ should_recalculate_difficulty(Height) ->
 calculate_difficulty(NewBlock, BlocksToCheckCount) ->
     CurrentTarget = NewBlock#block.target,
     CurrentRate = get_current_rate(NewBlock, BlocksToCheckCount),
```

Comment:
@akorosmezey  Please add TODO comment on current "rate" being positive.

This line identifies relevant two headers for computation, extracts time, computes time interval between oldest block and newest block. Depending on constraints on fields of header (depending also on other ticket re clock skew) and validation of such field, oldest block may have timestamp equal or newer (!!!) than newest block.

Reply:
@akorosmezey  Thanks for addressing this in `mining_rate_between_blocks`